### PR TITLE
added link for the examples of envFrom

### LINKED ISSUE #40572: “Define Environment Variables for a Container” does not show example for using `envFrom`
It is written ([here](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/)):

> This page explains how to use env.

What about `envFrom` example?

### PR DESCRIPTION
**Pull request by KhushPatibandha**
resolved #40572 by adding link to the examples for envFrom

### CODE REVIEW

**Review by dipesh-rawat — COMMENTED** [review]
The suggested change is not correct. The added link has new line character which is rendering the link wrong (Refer deploy preview: [deploy-PR-40579-define-environment-variable-container](https://deploy-preview-40579--kubernetes-io-main-staging.netlify.app/docs/tasks/inject-data-application/define-environment-variable-container/)). Also, it would be nice to use **_good commit messages_** :)

**I have put my thoughts/suggestions below :**
_(Apologies for providing lengthy comment)_

**TLDR** 
Though the suggested link demonstrates the use of `envFrom`, but to ensure clarity and completeness, it would be great to have all the necessary information in this task page. IMO, the best solution would be to add a new example showcasing the usage of `envFrom`. The current `env` example could be placed under a subsection let's say **'Setting container environment variable with env'**, and a new section titled **'Setting container environment variables with envFrom'** could be added. 
In this new section, we can use a similar example as the `env` example to highlight the differences between the two methods. This approach would provide practical use cases for both methods, enabling users to understand how and when to use them effectively.

For the new example for `envFrom` we would require the following: 
- File: **pods/inject/configmap-envFrom.yaml** 

```
 apiVersion: v1
kind: ConfigMap
metadata:
  name: envfrom-configmap
data:
  DEMO_GREETING: "Hello from the environment"
  DEMO_FAREWELL: "Such a sweet sorrow"
```

- File: **pods/inject/envFrom.yaml** 
```
apiVersion: v1
kind: Pod
metadata:
  name: envfrom-demo
  labels:
    purpose: demonstrate-envfrom
spec:
  containers:
  - name: envar-demo-container
    image: gcr.io/google-samples/node-hello:1.0
    envFrom:
    - configMapRef:
        name: envfrom-configmap
```

These yamls would need to be created and placed in a new set of manifest files ([here](https://github.com/kubernetes/website/tree/main/content/en/examples/pods/inject)) for the purpose of using the `codenew` shortcode to display them. The next step would be to provide instructions on how to create them and what output is expected.

**Shorter Version**
The preferred fix would look like below:  
// After the line 'You can read more about ConfigMap and Secret.’

**-  Setting container environment variables with env**
/* Existing examples goes here */

**- Setting container environment variables with envFrom**  
New content that needs to be written as part of this PR showing use of envFrom

**Comment by KhushPatibandha** [comment]
> 

Hey, @dipesh-rawat thank you for your feedback. I would surely look into it and try to incorporate all the suggestion that you have made.

**Comment by KhushPatibandha** [comment]
@dipesh-rawat I made the changes you suggested but I am still not confident that if this would work. So I just pushed the changes to show what I have worked on. I know these changes are not up to the standard but I would like some feedback.

File: content/en/examples/pods/inject/configmap-envFrom.yaml
Comment (sftim) [inline content/en/examples/pods/inject/configmap-envFrom.yaml:1]: Please use all lowercase for the filename.

File: content/en/examples/pods/inject/envFrom.yaml
Comment (sftim) [inline content/en/examples/pods/inject/envFrom.yaml:1]: Please use all lowercase for the filename.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 1]
```diff
@@ -80,6 +80,49 @@ Pod:
...
+{{< codenew file="pods/inject/configmap-envFrom.yaml" >}}
+
+Now as the kind: ConfigMap is created we need to 
+reference it to a pod. 
+
+{{< codenew file="pods/inject/envFrom.yaml" >}}
+
+You can also use Secret instead of ConfigMap for the configuration
+
+```
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:97]: nit: could append `yaml` here for syntax highlighting.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 2]
```diff
@@ -80,6 +80,49 @@ Pod:
...
+  name: mysecret
+type: Opaque
+data:
+  DB_HOST: bXlkb2Joc3RhcnQ= # Base64-encoded value for mydbhost
+  DB_PORT: NTQzMiA= # Base64-encoded value for 5432
+  DB_NAME: bXlkYm5hbWU= # Base64-encoded value for mydbname
+
+```
+
+Now as the Kind: Secret is created we need to reference it to a pod.
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:110]:
```suggestion
If you created that Secret, you could reference it in a bare Pod:
```

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 3]
```diff
@@ -80,6 +80,49 @@ Pod:
...
+  name: mypod
+spec:
+  containers:
+  - name: mycontainer
+    image: myimage
+    envFrom:
+    - secretRef:
+        name: mysecret
+```
+
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:125]:
```suggestion

## Interactions between environment variables

The environment variables you set using the `env` or `envFrom` field
override any environment variables specified in the container image.

Environment variables may reference each other, however ordering is important.
Variables making use of others defined in the same context must come later in
the list. Similarly, avoid circular references.
```

**Review by sftim — COMMENTED** [review]
Try these tweaks.

What do you think?

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md [hunk 4]
```diff
@@ -80,6 +80,49 @@ Pod:
...
 {{< note >}}
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:126]: Try replacing the two existing notes here with:
```markdown

## Interactions between environment variables

The environment variables you set using the `env` or `envFrom` field
override any environment variables specified in the container image.

Environment variables may reference each other, however ordering is important.
Variables making use of others defined in the same context must come later in
the list. Similarly, avoid circular references.
```

File: content/en/examples/pods/inject/envFrom.yaml
Comment (KhushPatibandha) [reply content/en/examples/pods/inject/envFrom.yaml:1]: so I should avoid using camel casing?

Comment (sftim) [reply content/en/examples/pods/inject/envFrom.yaml:1]: Yes. Use all-lowercase for filenames in this repo.

File: content/en/docs/tasks/inject-data-application/define-environment-variable-container.md (hunk 1 above)
Comment (KhushPatibandha) [reply content/en/docs/tasks/inject-data-application/define-environment-variable-container.md:97]: done

File: content/en/examples/pods/inject/envFrom.yaml
Comment (KhushPatibandha) [reply content/en/examples/pods/inject/envFrom.yaml:1]: so I've changed file names that I created while working on this issue, but I see 1 file with capital letter which is (\content\en\examples\pods\inject\pod-secret-envFrom.yaml) This is the file which got camel casing in it. So do you want this changed as well?

**Comment by KhushPatibandha** [comment]
checks are failing, what did I do wrong ?

**Comment by KhushPatibandha** [comment]
@sftim why are my checks failing?

**Comment by natalisucks** [comment]
@KhushPatibandha Could you try to push an empty commit to re-trigger the checks?

**Comment by Ritikaa96** [comment]
Hi @KhushPatibandha Please restart netlify checks by force pushing an empty commit .

**Comment by sftim** [comment]
/label tide/merge-method-squash

If you can, please squash the commits on your branch @KhushPatibandha. You should also verify that these changes preview OK locally.

**Comment by divya-mohan0209** [comment]
Hello @KhushPatibandha , since there have been no comments/commits on this PR despite reminders, I'll be closing it. Please note that if you find the bandwidth to address the suggestions and work on it again, feel free to issue the /reopen command.

/close
